cmd/admin-handlers.go

```diff
@@ -645,15 +865,15 @@ func (a adminAPIHandlers) UpdateCredentialsHandler(w http.ResponseWriter,
 	// Update local credentials in memory.
 	globalServerConfig.SetCredential(creds)
 
-	if err = saveServerConfig(objectAPI, globalServerConfig); err != nil {
+	if err = saveServerConfig(ctx, objectAPI, globalServerConfig); err != nil {
 		writeErrorResponseJSON(w, toAdminAPIErrCode(err), r.URL)
 		return
 	}
 
 	// Notify all other Minio peers to update credentials
 	for host, err := range globalNotificationSys.LoadCredentials() {
 		reqInfo := (&logger.ReqInfo{}).AppendTags("peerAddress", host.String())
```

Comment:
if this tag gets appended, then when the actual error occurs, we will be logging hosts which did not fail.

Reply:
> if this tag gets appended, then when the actual error occurs, we will be logging hosts which did not fail.

I didn't understand your point, reqInfo is set in each iteration, why will host will be wrong when there is an error in the current iteration ?

Reply:
Since we are setting, new `ReqInfo`, this will not append tags. But still, I think this should be within an error check, as later if someone is reusing the `ReqInfo` attached to this `ctx`, it will have this tag attached to it. But then this is a problem, even if there is an error.

Reply:
``for host, err := range globalNotificationSys.LoadCredentials() {
    if err != nil {
         logger.GetReqInfo(ctx).AppendTags("peerAddress", host.String())
         logger.LogIf(ctx, err)
   }

```